clients/da-vinci-client/src/main/java/com/linkedin/davinci/stats/ingestion/heartbeat/HeartbeatMonitoringService.java

```diff
@@ -339,6 +340,47 @@ public void updateLagMonitor(
     }
   }
 
+  /**
+   * Get maximum heartbeat lag from all regions for a given LEADER replica.
+   * @return Max leader heartbeat lag, or Long.MAX_VALUE if any region's heartbeat is unknown.
+   */
+  public long getReplicaLeaderHeartbeatLag(String storeName, int version, int partitionId) {
+    Map<String, HeartbeatTimeStampEntry> replicaTimestampMap =
+        getLeaderHeartbeatTimeStamps().getOrDefault(storeName, Collections.emptyMap())
+            .getOrDefault(version, Collections.emptyMap())
+            .getOrDefault(partitionId, Collections.emptyMap());
+    if (replicaTimestampMap.isEmpty()) {
+      return Long.MAX_VALUE;
+    }
+    long minTimestamp = Long.MAX_VALUE;
+    for (Map.Entry<String, HeartbeatTimeStampEntry> entry: replicaTimestampMap.entrySet()) {
```

Comment:
Do we need filtering in here somehow for the inc push colo?

Reply:
No, we need to track all subscribed to consume regions and make sure they are all caught up for ready-to-serve check. Otherwise it will be semantic confusing for user as they are still seeing nothing but the replica says it is fully caught up. 
This is the same logic in the offset lag based calculation.

Reply:
Synced offline. We reached an alignment that sepRT region should not be tracked for RTS actually. If it is tracked it could delay the node to come up whenever there comes a big ingestion from inc push.